src/DataFramesMeta.jl

```diff
@@ -387,18 +386,32 @@ function transform(d::Union{AbstractDataFrame, AbstractDict}; kwargs...)
     return result
 end
 
+
 function transform(g::GroupedDataFrame; kwargs...)
     result = DataFrame(g)
-    idx2 = cumsum(Int[size(g[i],1) for i in 1:length(g)])
-    idx1 = [1; 1 + idx2[1:end-1]]
-    for (k, v) in kwargs
-        first = v(g[1])
-        result[k] = Array{eltype(first)}(size(result, 1))
-        result[idx1[1]:idx2[1], k] = first
-        for i in 2:length(g)
-            result[idx1[i]:idx2[i], k] = v(g[i])
+
+    function spread_scalar(x::Vector, obs_in_group::Int)
+        if length(x) == obs_in_group 
+            return x 
+        else
+            throw(DimensionError("If a function returns a vector, the result " * 
+                                 "must have the same length as the groups it " *
+                                 "operates on"))
         end
     end
+
+    function spread_scalar(x::CategoricalArrays.CategoricalValue, obs_in_group::Int)
```

Comment:
Actually, `(x for x in 1:obs_in_group)` isn't indexable, so my suggestion won't work. I guess the best solution is to have an `if x isa AbstractArray` branch in the `for` loop below, rather than dispatching on methods.